flow/scheduling/base.py

```diff
@@ -10,16 +10,41 @@ class JobStatus(enum.IntEnum):
     """Classifies the job's execution status."""
 
     unknown = 1
+    """Unknown cluster job status."""
+
     registered = 2
+    """The cluster job is registered with the scheduler, but no other status is known."""
+
     inactive = 3
+    """The cluster job is inactive.
+
+    This includes states like completed, cancelled, or timed out.
+    """
+
     submitted = 4
+    """The cluster job has been submitted.
+
+    Note that this state is never returned by a scheduler, but is an assumed
+    state immediately after a cluster job is submitted.
+    """
+
     held = 5
+    """The cluster job is held."""
+
     queued = 6
+    """The cluster job is queued."""
+
     active = 7
+    """The cluster job is actively running."""
+
     error = 8
-    dummy = 127
-    # All user statuses are >= 128.
+    """The cluster job is in an error or failed state."""
+
+    placeholder = 127
+    """A placeholder state that is used internally."""
```

Comment:
I would specify the purpose of this status here since it exists for only one purpose and we don't want to make it seem like something random that just gets added.

Reply:
```suggestion
    """A placeholder state that is used for status rendering when no operations are eligible."""
```